Inmobi Library_files/script.js

```diff
@@ -51,21 +65,34 @@ function loadInterstitialAd() {
   adInstances.push(adInstance);
 }
 
+function check() {
+  const form = document.getElementById("myform");
+  if(form.checkValidity()){
+    req_no = document.getElementById("noreq").value;
+    console.log("no of req: ",req_no)
+    req_ctr = 0;
+    document.getElementById("adType").value === 'banner' ? loadBanner() : loadInterstitialAd()
+  }
+}
 function loadBanner() {
   updateBannerConatiner()
   const config = getMockConfig();
-  //console.log(config)
-  const adInstance = mesonWebClient.getBannerAd(config);
+  var adInstance = mesonWebClient.getBannerAd(config);
+  console.log(adInstance)
   adInstance?.load();
-
+   req_ctr ++ ;
   adInstance?.onAdLoadSucceeded((requestId) => {
     console.log('onAdLoadSucceeded!!', requestId);
     // Optionally, you can call the showInterstitialAd function here if desired
-    document.getElementById("console-log").innerHTML += `Banner :  Ad Loaded!!! Yipeee!! ${JSON.stringify(requestId)}<br>`;
+    document.getElementById("console-log").innerHTML += `Banner :  Ad Loaded!!! Yipeee!! ${JSON.stringify(requestId)}<br>
+    Number of requests tried: ${req_ctr}<br>`;
+    req_ctr = 0;
   });
 
    adInstance?.onAdLoadFailed((error) => {
+    console.log('onAdLoadFailed!');
       document.getElementById("console-log").innerHTML += `Banner : Load failed received on client side: ${JSON.stringify(error)}<br>`;
+      req_ctr < req_no && loadBanner()
     });
```

Comment:
create a new method for retryAd(). Add all the logic there to load the next ad or not.